tests/e2e_tests/test_data_compatibility.py

```diff
@@ -0,0 +1,150 @@
+        
+        print(f"Downloading compatibility data for {version}...")
+
+        try:
+            with requests.get(url, stream=True, timeout=(10, 300)) as response:
+                response.raise_for_status()
+                with open(compatibility_file, 'wb') as f:
+                    for chunk in response.iter_content(chunk_size=1024 * 1024):
+                        if chunk:
+                            f.write(chunk)
+        except requests.exceptions.RequestException as e:
+            pytest.skip(f"Could not download compatibility data for {version}: {e}")
+        
+        return compatibility_file
+
+    @staticmethod
+    def _extract_storage_data(compatibility_file: Path, storage_test_dir: Path):
+        """Extract storage data from compatibility archive."""
+        extract_archive(compatibility_file, storage_test_dir, cleanup_archive=True)
+        
+        # Extract nested storage archive
+        storage_archive = storage_test_dir / "storage.tar.bz2"
+        if storage_archive.exists():
+            extract_archive(storage_archive, storage_test_dir, cleanup_archive=True)
+        
+        return storage_test_dir / "storage"
+
+    @staticmethod
+    def _extract_snapshot_data(storage_test_dir: Path):
+        """Extract snapshot data."""
+        snapshot_gz = storage_test_dir / "full-snapshot.snapshot.gz"
+        
+        if snapshot_gz.exists():
+            extract_archive(snapshot_gz, storage_test_dir, cleanup_archive=True)
+            return storage_test_dir / "full-snapshot.snapshot"
+        
+        return None
+
+    @staticmethod
+    def _check_collections(host: str, port: int) -> bool:
+        """Check that all collections are loaded properly."""
+        client = ClientUtils(host=host, port=port)
+
+        try:
+            collections = client.list_collections_names()
+        except Exception as e:
+            print(f"Error listing collections: {e}")
+            return False
+
+        expected = set(TestStorageCompatibility.EXPECTED_COLLECTIONS)
+        found = set(collections)
+        missing = expected - found
+        if missing:
+            print(f"Missing expected collections: {sorted(missing)}")
+            return False
+
+        for collection in expected:
+            try:
+                collection_info = client.get_collection_info_dict(collection)
+                if collection_info["status"] != "ok":
+                    print(f"Collection {collection} returned status {collection_info['status']}")
+                    return False
+            except Exception as error:
+                print(f"Failed to get collection info for {collection}: {error}")
+                return False
+        return True
+
+
+    @pytest.mark.parametrize("version", [PREV_PATCH_VERSION, PREV_MINOR_VERSION])
+    def test_storage_compatibility(self, docker_client, qdrant_image, temp_storage_dir, version, qdrant_container_factory):
```

Comment:
Is it mandatory to use a Docker build now?
I prefer working with a simple binary when reproducing issues TBH